src/mechanics/cell/Cell.h

```diff
@@ -78,5 +84,6 @@ class Cell : public CellInterface, UniqueId<Cell>
 private:
     const ElementCollection& mElements;
     const IntegrationTypeBase& mIntegrationType;
+    const int mId;
```

Comment:
I would generally not use `const` here. This prevents some compiler generated trivial copy/move stuff [How would you manually write a copy assignment?]. If you keep this data private, no one should be able to modify it from outside. The other data members remove the trivial assignment methods anyways, so it does not matter much here, but I find it interesting. Some more on this topic [here](http://www.drdobbs.com/the-problem-with-const-data-members/184403306).

Reply:
You might be right, that const is not necessary if the data is not exposed. On the other hand, do we want to copy assign something with an ID? Afterwards you have two objects with the same ID. Do we want that?

Reply:
You are totally right. Maybe we should even explicitly delete copy/move ctor/assign. I just wanted to write it here, because this _make it as `const` as possible_ is not always the best thing to do.

Reply:
I think moving is fine and should be allowed. However, since there are also references stored in the class, assignments couldn't be done anyways without also changing them to pointers. So if nobody has any complains, I would delete the copy ctor, and both assignment operators for the cell.